dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt

```diff
@@ -235,13 +235,21 @@ internal open class RumViewScope(
                     actionId = null
                 ),
                 applyOnlyIf = { currentContext ->
-                    if (currentContext.viewId == this.viewId) {
-                        true
-                    } else {
-                        sdkLogger.debugWithTelemetry(
-                            RUM_CONTEXT_UPDATE_IGNORED_AT_STOP_VIEW_MESSAGE
-                        )
-                        false
+                    when {
+                        currentContext.sessionId != this.sessionId -> {
```

Comment:
good catch

Reply:
If the session id is different then it might mean that another view is active, which you could be erasing here. What's your logic for doing this check?

Reply:
Normally it shouldn't be another view at this point, because the current one should be stopped first (if we are saying that `stopView` is called). If `stopView` is not called for current view and another view starts (via `startView`), then `stopped` flag will be `true` after `onStartView` with event having key of another view, so we [won't even go in this logical branch](https://github.com/DataDog/dd-sdk-android/blob/75882597ab6c730cb1e98e8d71e1367fb3c6acf4/dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt#L228).

Anyway, such override was a case even before, when `GlobalRum.updateRumContext` was called in `onStopView` without such verification, so at least we are not making things worse.